Reworks examples in README and introduces WASI (#387)
tetratelabs/wazero: README.md

# wazero

WebAssembly is a way to safely run code compiled in other languages. Runtimes

> execute WebAssembly Modules (Wasm), which are most often binaries with a `%.wasm`

Author: ```suggestion
execute WebAssembly Modules (Wasm), which are most often binaries with a `.wasm`
```

}
```

> Note: While the [source for this](tests/engine/testdata/fac.wat) is in the

Reviewer: oh I realized that `fac.wat` is no longer existent :D

WebAssembly 1.0 (20191205) Text Format, it could have been written in another

> language that compiles to (targets) WebAssembly, such as AssemblyScript/C/C++/Rust/TinyGo/Zig.

Reviewer: I would prefer 
```suggestion
language that compiles to (targets) WebAssembly, such as AssemblyScript, C, C++, Rust, TinyGo or Zig.
```

## Deeper dive

The former example is a pure function. While a good start, you probably are
wondering how to do something more realistic, like read a file. WebAssembly
Modules (Wasm) are sand-boxed similar to containers. They can't read anything
on your machine unless you explicitly allow it.

System access is defined by an emerging specification called WebAssembly
System Interface ([WASI](https://github.com/WebAssembly/WASI)). WASI defines
how WebAssembly programs interact with the host embedding them. For example,
WASI defines functions for reading the time, or a random number.

> This repository include several [examples](examples) use system interfaces,

Author: ```suggestion
This repository include several [examples](examples) that expose system interfaces,
```
